packages/studio-plugin/src/parsers/helpers/TypeNodeParsingHelper.ts

```diff
@@ -137,12 +137,49 @@ export default class TypeNodeParsingHelper {
     }
   }
 
+  private static getInheritedShape(
+    shapeDeclaration: InterfaceDeclaration | TypeLiteralNode,
+    parseTypeReference: (identifier: string) => ParsedType | undefined
+  ): ParsedShape {
+    if (!shapeDeclaration.isKind(SyntaxKind.InterfaceDeclaration)) {
+      return {};
+    }
+
+    const heritageClause = shapeDeclaration.getHeritageClauseByKind(
+      SyntaxKind.ExtendsKeyword
+    );
+    if (!heritageClause) {
+      return {};
+    }
+
+    const inheritedTypes = heritageClause.getTypeNodes();
+    return inheritedTypes.reduce((inheritedShape, inheritedType) => {
+      const typeName = StaticParsingHelpers.getEscapedName(
+        inheritedType.getExpression()
+      );
+      const parsedType = parseTypeReference(typeName);
+      if (parsedType?.kind === ParsedTypeKind.Object) {
+        return {
+          ...inheritedShape,
+          ...parsedType.type,
+        };
+      } else {
+        throw new Error(
```

Comment:
In the case of parsing a Component's Prop Interface, if this `Error` is thrown, do we gracefully handle? That is, do we put the Component in the Error State in the UI?

Reply:
Yes, as with other parsing errors, this puts the component in an error state. This means in the UI we remove the component from the Add Element list and display any page's usages of the component in red with a tooltip (both in the component tree and the preview).

<img width="1293" alt="Screenshot 2023-10-05 at 9 43 33 AM" src="https://github.com/yext/studio/assets/88398086/0ccec07d-68d3-43de-8230-7cbe06679182">